api/catalog/api/utils/validate_images.py

```diff
@@ -57,24 +81,13 @@ def validate_images(query_hash, start_slice, results, image_urls):
             to_verify[url] = idx
     logger.debug(f"len(to_verify)={len(to_verify)}")
 
-    reqs = (
-        grequests.head(
-            url, headers=HEADERS, allow_redirects=False, timeout=2, verify=False
-        )
-        for url in to_verify.keys()
-    )
-    verified = grequests.map(reqs, exception_handler=_validation_failure)
+    verified = _make_head_requests(to_verify.keys())
+
     # Cache newly verified image statuses.
     to_cache = {}
-    # relies on the consistenct of the order returned by `dict::keys()` which
-    # is safe as of Python 3 dot something
-    for idx, url in enumerate(to_verify.keys()):
+    for result in verified:
+        url, status = result
```

Comment:
Would it be possible to just unpack on the iteration?
```suggestion
    for url, status in verified:
```

Reply:
Yup, changed the whole statement to a comprehension in https://github.com/WordPress/openverse-api/pull/1027/commits/582d563ada7def1b8ded7b0c80d3b994fe22deda